backend/service/k8s/deployment.go

```diff
@@ -101,12 +101,144 @@ func ProtoForDeploymentSpec(deploymentSpec appsv1.DeploymentSpec) *k8sapiv1.Depl
 			resourceRequests[string(res)] = quantity.String()
 		}
 
+		LivenessProbeObject := &k8sapiv1.Probe{}
+		ReadinessProbeObject := &k8sapiv1.Probe{}
+		if container.LivenessProbe != nil {
+			HandlerObj := &k8sapiv1.Probe{}
+
+			if container.LivenessProbe.ProbeHandler.HTTPGet != nil {
+				LivenessProbeHTTPHeaders := make([]*k8sapiv1.HTTPHeader, 0, len(container.LivenessProbe.ProbeHandler.HTTPGet.HTTPHeaders))
+				for _, value := range container.LivenessProbe.ProbeHandler.HTTPGet.HTTPHeaders {
+					UniqueLivenessHeader := &k8sapiv1.HTTPHeader{
+						Name:  &value.Name,
+						Value: &value.Value,
+					}
+					LivenessProbeHTTPHeaders = append(LivenessProbeHTTPHeaders, UniqueLivenessHeader)
+				}
+
+				LivenessProbeHTTPObject := &k8sapiv1.HTTPGetAction{
+					Path:        container.LivenessProbe.ProbeHandler.HTTPGet.Path,
+					Port:        container.LivenessProbe.ProbeHandler.HTTPGet.Port.IntVal,
+					Host:        container.LivenessProbe.ProbeHandler.HTTPGet.Host,
+					Scheme:      string(container.LivenessProbe.ProbeHandler.HTTPGet.Scheme),
+					HttpHeaders: LivenessProbeHTTPHeaders,
+				}
+				HandlerObj.Handler = &k8sapiv1.Probe_HttpGet{
+					HttpGet: LivenessProbeHTTPObject,
+				}
+			}
+
+			if container.LivenessProbe.ProbeHandler.Exec != nil {
+				LivenessProbeExec := &k8sapiv1.ExecAction{
+					Command: container.LivenessProbe.ProbeHandler.Exec.Command,
+				}
+				HandlerObj.Handler = &k8sapiv1.Probe_Exec{
+					Exec: LivenessProbeExec,
+				}
+			}
+
+			if container.LivenessProbe.ProbeHandler.TCPSocket != nil {
+				LivenessProbeTCPSocket := &k8sapiv1.TCPSocketAction{
+					Port: container.LivenessProbe.ProbeHandler.TCPSocket.Port.IntVal,
+					Host: container.LivenessProbe.ProbeHandler.TCPSocket.Host,
+				}
+				HandlerObj.Handler = &k8sapiv1.Probe_TcpSocket{
+					TcpSocket: LivenessProbeTCPSocket,
+				}
+			}
+
+			if container.LivenessProbe.ProbeHandler.GRPC != nil {
+				LivenessProbeGRPC := &k8sapiv1.GRPCAction{
+					Port:    container.LivenessProbe.ProbeHandler.GRPC.Port,
+					Service: *container.LivenessProbe.ProbeHandler.GRPC.Service,
+				}
+				HandlerObj.Handler = &k8sapiv1.Probe_Grpc{
+					Grpc: LivenessProbeGRPC,
+				}
+			}
```

Comment:
nit: maybe we should use a switch here? only one of these can be configured at a time.

Reply:
I try but I think it is not possible to switch with this type of structure I moved to a new method and reduce the amount of code